pkg/kallsyms/kallsyms_test.go

```diff
@@ -224,3 +226,258 @@ func BenchmarkKernelSymbolAddressPreLoading(b *testing.B) {
+	}{
+		{
+			name:     "exact_match_start",
+			ip:       0x0,
+			expected: "first_symbol",
+		},
+		{
+			name:     "before_first_symbol",
+			ip:       0x0,
+			expected: "first_symbol",
+		},
+		{
+			name:     "between_symbols",
+			ip:       0x1500,
+			expected: "second_symbol",
+		},
+		{
+			name:     "exact_match_end",
+			ip:       0x3000,
+			expected: "fourth_symbol",
+		},
+		{
+			name:     "after_last_symbol",
+			ip:       0x4000,
+			expected: "fourth_symbol",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := kAllSyms.LookupByInstructionPointer(tt.ip)
+			require.Equal(t, tt.expected, result)
+		})
+	}
+}
+
+func TestKernelSymbolAddressFromReaderErrors(t *testing.T) {
+	tests := []struct {
+		name        string
+		input       string
+		symbol      string
+		expectedErr string
+	}{
+		{
+			name:        "invalid_line_no_type",
+			input:       "ffffffffb4231f40",
+			symbol:      "test_symbol",
+			expectedErr: "parsing line \"ffffffffb4231f40\": no type",
+		},
+		{
+			name:        "invalid_line_no_symbol",
+			input:       "ffffffffb4231f40 T",
+			symbol:      "test_symbol",
+			expectedErr: "parsing line \"ffffffffb4231f40 T\": no symbol",
+		},
+		{
+			name:        "invalid_address",
+			input:       "INVALID T test_symbol",
+			symbol:      "test_symbol",
+			expectedErr: "parsing line \"INVALID T test_symbol\": invalid address",
+		},
+		{
+			name:        "duplicate_symbol",
+			input:       "ffffffffb4231f40 T test_symbol\nffffffffb4231f41 T test_symbol",
+			symbol:      "test_symbol",
+			expectedErr: "duplicate found at address",
+		},
+		{
+			name:        "symbol_not_found",
+			input:       "ffffffffb4231f40 T other_symbol",
+			symbol:      "test_symbol",
+			expectedErr: "file does not exist",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			reader := strings.NewReader(tt.input)
+			_, _, err := kernelSymbolAddressFromReader(reader, tt.symbol)
+			require.Error(t, err)
+			require.Contains(t, err.Error(), tt.expectedErr)
+		})
+	}
+}
+
+func TestKernelSymbolAddressFromReaderModuleSuccess(t *testing.T) {
+	input := "ffffffffc1234567 T test_symbol\t[test_module]"
+	reader := strings.NewReader(input)
+	addr, mod, err := kernelSymbolAddressFromReader(reader, "test_symbol")
+	require.NoError(t, err)
+	require.Equal(t, uint64(0xffffffffc1234567), addr)
+	require.Equal(t, "test_module", mod)
+}
+
+func TestSymbolExistsWithEmptySymbol(t *testing.T) {
+	exists := SymbolExists("")
+	require.False(t, exists)
+}
+
+// Test for parsing various module formats
+func TestKernelSymbolAddressFromReaderModuleFormats(t *testing.T) {
+	tests := []struct {
+		name     string
+		input    string
+		symbol   string
+		wantMod  string
+		wantAddr uint64
+	}{
+		{
+			name:     "no_module",
+			input:    "ffffffffc1234567 T test_symbol",
+			symbol:   "test_symbol",
+			wantMod:  "",
+			wantAddr: 0xffffffffc1234567,
+		},
+		{
+			name:     "with_module",
```

Comment:
This test is kind of the same as `module_with_with_underscore` since the modile name has an underscore here too